requirements-optional.txt

```diff
@@ -0,0 +1,31 @@
+# Optional dependencies of Terra that can (mostly) reliably be installed with
+# `pip`.  This file is still called `requirements-optionals.txt` just to match
+# standard pip conventions, even though none of these are required.
+#
+# If updating this, you probably want to update `qiskit.utils.optionals` too.
+
+qiskit-aer
+qiskit-toqm>=0.0.4; platform_machine != 'aarch64' or platform_system != 'Linux'
+python-constraint>=1.4
+cplex; python_version < '3.11'
+cvxpy
+docplex
+fixtures
```

Comment:
I'm wondering if we want to do some grouping of requirements here like you did for requirements-dev.txt.

I think it would be useful here to map this for people looking at specific features. I did a double take when I first looked at this and thought to myself fixtures is a test only requirement.

Reply:
I've made some rudimentary sectionining in 8f841901e - I can add more if you think it's worthwhile.